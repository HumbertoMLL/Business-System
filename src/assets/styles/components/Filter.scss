@import '../Vars.scss';

.section__filter {
  grid-column: 1;
  display: flex;
  flex-direction: column;
  align-items: center;

&--item {
    display: flex;
    flex-direction: column;
    margin: 1em 0 1em 1em;
    border-radius: 4px;

& h2 {
    align-self: center;
    font-size: 26px;
    letter-spacing: 2px;
    font-weight: 700 ;
    font-family: $fuente;
    margin: 0;

}
}
&--options {
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 20em;
  background-color:$color-suave;
  border-radius: 4px;
  list-style: none;
  padding: 0;
  & h3 {
      font-size: 24px;
      font-family: $fuente;
      font-weight: 400;
      text-align: center;
      margin: 0;
      margin-top: 16px;
  }
  & h4 {
    align-self: flex-start;
    font-size: 20px;
    text-align: justify;
    font-family: $fuente;
    margin-top: 16px;
    margin-left: 8px;
    margin-bottom: 16px;
  }

& button {
    width: 14em;
    height: 3em;
    border-radius: 0;
    font-family: $fuente;
    font-size: 18px;
    letter-spacing: 2px;
    background: transparent;
    border: none;
    margin-bottom: 8px;

&:hover {
  background-color: $color-fuerte;
}
&:focus {
    outline: none;
}
}

}

}