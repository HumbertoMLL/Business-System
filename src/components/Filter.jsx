import React from 'react';
import '../assets/styles/components/Filter.scss';

const Filter = () => (
  <section className='section__filter'>
    <div className='section__filter--item'>
      <h2>Lista de empleados</h2>

      <ul className='section__filter--options'>
        <h3> Todos los empleados </h3>
        <h4> Proyecto </h4>
        <li>
          <button type='button'> Wizline </button>
        </li>
        <li>
          <button type='button'> Tesla </button>
        </li>
        <li>
          <button type='button'> Arduino </button>
        </li>
        <li>
          <button type='button'> Gitlab </button>
        </li>
        <li>
          <button type='button'> Globant </button>
        </li>
        <li>
          <button type='button'> Platzi</button>
        </li>
      </ul>
    </div>
  </section>
);

export default Filter;
