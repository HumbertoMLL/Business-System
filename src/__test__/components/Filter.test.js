import React from 'react';
import { mount } from 'enzyme';
import { create } from "react-test-renderer";
import Filter from '../../components/Filter';

describe('<Filter />', () => {
  test('Render del componente Filter', () => {
    const filter = mount(<Filter />);
    expect(filter.length).toEqual(1);
  });
});

describe('Filter Snapshot', () => {
  test('Comprobar la UI del componente Filter', () => {
    const filter = create(<Filter />);
    expect(filter.toJSON()).toMatchSnapshot();
  });
});
//Para actualizar el snapshot cuando actualizamos el html
//Usamos el comando en la terminal
//jest --updateSnapshot
