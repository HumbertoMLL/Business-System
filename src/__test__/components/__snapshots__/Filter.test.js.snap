// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Snapshot Comprobar la UI del componente Filter 1`] = `
<section
  className="section__filter"
>
  <div
    className="section__filter--item"
  >
    <h2>
      Lista de empleados
    </h2>
    <ul
      className="section__filter--options"
    >
      <h3>
         Todos los empleados 
      </h3>
      <h4>
         Proyecto 
      </h4>
      <li>
        <button
          type="button"
        >
           Wizline 
        </button>
      </li>
      <li>
        <button
          type="button"
        >
           Tesla 
        </button>
      </li>
      <li>
        <button
          type="button"
        >
           Arduino 
        </button>
      </li>
      <li>
        <button
          type="button"
        >
           Gitlab 
        </button>
      </li>
      <li>
        <button
          type="button"
        >
           Globant 
        </button>
      </li>
      <li>
        <button
          type="button"
        >
           Platzi
        </button>
      </li>
    </ul>
  </div>
</section>
`;
